src/huggingface_hub/utils/auth.py

```diff
@@ -0,0 +1,31 @@
+import sys
+from base64 import b64encode
+from typing import Optional
+
+
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
+
+
+class AuthHeaders(TypedDict):
+    authorization: str
```

Comment:
Nice ! Why not use a dataclass here ?

In your code you assume that the `authorization` field always exist in an instance of `AuthHeaders`, but it's ont the case for an empty `AuthHeaders()` for example. So if you want to fix that, I guess you can either use `auth.get("authorization")` instead of `auth["authorization"]`, or switch to a dataclass (as you want)